framework/src/org/checkerframework/common/value/ValueTransfer.java

```diff
@@ -1324,6 +1326,58 @@ private void addAnnotationToStore(CFStore store, AnnotationMirror anno, Node nod
                 transferResult, firstNode, secondNode, firstValue, secondValue, notEqualTo);
     }
 
+    @Override
+    protected void processConditionalPostconditions(
+            MethodInvocationNode n,
+            ExecutableElement methodElement,
+            Tree tree,
+            CFStore thenStore,
+            CFStore elseStore) {
+        // For String.startsWith(String) and String.endsWith(String), refine the minimum length
+        // of the receiver to the minimum length of the argument.
+        // Basically, MinLen annotations are transferred from the argument to the receiver.
+        // However, a polymorphic qualifier cannot be used to express this, because
```

Comment:
Thanks for the explanation.   Issue  https://github.com/kelloggm/checker-framework/issues/56 isn't very clear.  The proposal was to use the polymorphic qualifier in an `@EnsuresQualifierIf` annotation.    So `startsWith` would be annotated like:
```
@EnsuresQualifierIf( result = true, expression = "this", qualifier = PolyValue.class)
boolean startsWith(@PolyValue String arg);
```
However `@EnsuresQualifierIf` doesn't support polymorphic qualifiers, so the special casing you implemented was required.